base/compiler/ssair/inlining.jl

```diff
@@ -967,11 +982,11 @@ function analyze_method!(match::MethodMatch, argtypes::Vector{Any},
     mi = specialize_method(match; preexisting=true) # Union{Nothing, MethodInstance}
     if mi === nothing
         et = InliningEdgeTracker(state.et, invokesig)
-        return compileable_specialization(match, Effects(), et, info;
+        return compileable_specialization(match, override_effects === nothing ? Effects() : override_effects, et, info;
```

Comment:
I think we can do something like:
```suggestion
        return compileable_specialization(match, override_effects === nothing ? info_effects(nothing, match, state) : override_effects, et, info;
```
and get more precision.